File: AtCoder/ACP/BootCamp4b/Medium100_2/cpp/D_Dice_in_Line.cpp
#include <bits/stdc++.h>
#define MAX_N 200'000
using namespace std;

int N, K;
int P[MAX_N];
double sum[MAX_N + 1];

void solve() {
  for (int i = 0; i < N; ++i) {
    sum[i + 1] = sum[i] + (P[i] + 1) / 2.0;
  }
  double res = 0;
  for (int i = 0; i + K <= N; ++i) {
    res = max(res, sum[i + K] - sum[i]);
  }
  printf("%.7f\n", res);
}

int main() {
  cin >> N >> K;
  for (int i = 0; i < N; ++i) {
    cin >> P[i];
  }
  solve();
}

File: AtCoder/ACP/BootCamp4b/Medium100_2/python/D_Dice_in_Line.py
N, K = map(int, input().split())
P = list(map(int, input().split()))

sm = [0.0] * (N + 1)
for i in range(N):
    sm[i + 1] = sm[i] + (P[i] + 1) / 2
res = 0
for i in range(N - K + 1):
    res = max(res, sm[i + K] - sm[i])
print(f"{res:.7f}")
